### PR TITLE
Present Search Results - eui 89

### PR DESCRIPTION
**Pull request by praveenrajendran101**
**Before creating a pull request make sure that:**

- [ X] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-56

### Change description ###
Added search filter with end to end integration. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: api/lib/proxy.ts [hunk 1]
```diff
@@ -23,8 +23,9 @@ function setHeaders(req: EnhancedRequest) {
 
 export async function get(req: EnhancedRequest, res: express.Response, next: express.NextFunction) {
     let url = striptags(req.url)
-    url = '/data/' + url
-
+    url = req.baseUrl  + url
+    console.log(url)
```
Comment (adnanakgun) [inline api/lib/proxy.ts]: Is console.log wanted here?

File: src/styles.scss [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
-$govuk-frontend-toolkit-image-path: "/assets";
-$govuk-template-image-path: "/assets";
+// $fa-font-path: "/node_modules/font-awesome/fonts";
+// $govuk-frontend-toolkit-image-path: "/assets";
+// $govuk-template-image-path: "/assets";
 
-@import "govuk-elements";
+// @import "govuk-elements";
+
+@import './style/app.scss';
```
Comment (adnanakgun) [inline src/styles.scss]: I suppose commented out lines could be removed as they seemed to have been moved to app.scss

File: .vscode/launch.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (fazanki) [inline .vscode/launch.json]: don't think we need this to be committed not everybody is using VS Code...

File: src/assets/health.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (fazanki) [inline src/assets/health.json]: can u add a comment why is this here ?

File: src/style/_markdown.scss [hunk 5]
```diff
@@ -0,0 +1,47 @@
+@import 'node_modules/@hmcts/ccd-case-ui-toolkit/dist/cut-styles/components/typography';
```
Comment (fazanki) [inline src/style/_markdown.scss]: don't think we should me manging scc like this... it should be in import from node module... 
if u do decide to leave it please to do and speak to mo and andy for a solution

**Comment by fazanki** [comment]
can't see the search result or is it work in progress?

File: src/cases/store/entity.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+        this.id.push(o.id);
+        this[o.id] = o;
+      }
+    } else {
+      this.id = obj.id;
+      this[obj.id] = obj;
+    }
+  }
+}
+
```
Comment (fazanki) [inline src/cases/store/entity.ts]: ❤️

File: src/cases/store/reducers/search-filter.reducer.ts [hunk 7]
```diff
@@ -1,31 +1,31 @@
...
-
-export function reducer(
-  state = initialStateSearch,
-  action: fromCases.CaseSearchAction
-): SearchState {
+export function reducer( state , action: fromCases.CaseSearchAction): SearchState {
   switch (action.type) {
     case fromCases.APPLIED: {
-      const searchFilterCriteria = action.payload;
+      const metadataFields = new Entity(action.payload.metadataFields);
```
Comment (fazanki) [inline src/cases/store/reducers/search-filter.reducer.ts]: ❤️

File: .vscode/launch.json (hunk 3 above)
Comment (smogese) [reply .vscode/launch.json]: Theres a whole vscode section in .gitignore 

# IDE - VSCode
.vscode/*
!.vscode/settings.json
!.vscode/tasks.json
!.vscode/launch.json
!.vscode/extensions.json
.history/*

Is this not being read

**Comment by praveenrajendran101** [comment]
@smogese gitignore seems working for me
